junto-rust/zomes/core/code/src/expressions/indexing.rs

```diff
@@ -0,0 +1,266 @@
+                let dht_entry = hdk::get_entry(&address)?;
+
+                match dht_entry {
+                    Some(_value) => {
+                        hdk::api::link_entries(&address, expression, "expression")?;
+                        hdk::api::link_entries(&address, expression, query_param["value"].to_string())?;
+                        hdk::api::link_entries(&expression, &address, "time")?;
+                        hdk::api::link_entries(&expression, &address, "day")?;
+                        addressed_params[i].insert("address".to_string(), address.to_string());
+                    },
+                    None => {
+                        let address = hdk::utils::commit_and_link(&entry, context, "time")?;
+                        hdk::api::link_entries(context, &address, "day")?;
+                        hdk::api::link_entries(&address, expression, "expression")?;
+                        hdk::api::link_entries(&address, expression, query_param["value"].to_string())?;
+                        hdk::api::link_entries(&expression, &address, "time")?;
+                        hdk::api::link_entries(&expression, &address, "day")?;
+                        addressed_params[i].insert("address".to_string(), address.to_string());
+                    }
+                };
+            },
+            "Time:H" => {
+                let entry = Entry::App("time".into(), app_definitions::Time{time: query_param["value"].to_string(), 
+                                        parent: context.clone(), time_type: app_definitions::TimeType::Hour}.into()).into();
+                let address = hdk::entry_address(&entry)?;
+                let dht_entry = hdk::get_entry(&address)?;
+
+                match dht_entry {
+                    Some(_value) => {
+                        hdk::api::link_entries(&address, expression, "expression")?;
+                        hdk::api::link_entries(&address, expression, query_param["value"].to_string())?;
+                        hdk::api::link_entries(&expression, &address, "time")?;
+                        hdk::api::link_entries(&expression, &address, "hour")?;
+                        addressed_params[i].insert("address".to_string(), address.to_string());
+                    },
+                    None => {
+                        let address = hdk::utils::commit_and_link(&entry, context, "time")?;
+                        hdk::api::link_entries(&context, &address, "hour")?;
+                        hdk::api::link_entries(&address, expression, "expression")?;
+                        hdk::api::link_entries(&address, expression, query_param["value"].to_string())?;
+                        hdk::api::link_entries(&expression, &address, "time")?;
+                        hdk::api::link_entries(&expression, &address, "hour")?;
+                        addressed_params[i].insert("address".to_string(), address.to_string());
+                    }
+                };
+            },
+            "user" => {
```

Comment:
All of this is begging for an abstraction! Interested to hear how this goes and if we can make a pattern that is reusable in other hApps